src/record/partially_serialized.rs

```diff
@@ -3,20 +3,56 @@ use crate::prelude::*;
 use bytes::{Bytes, BytesMut};
 
 pub(crate) struct PartiallySerializedWriteResult {
-    bytes_written: u64,
+    blob_offset: u64,
     header_checksum: u32,
 }
 
 impl PartiallySerializedWriteResult {
-    pub(crate) fn bytes_written(&self) -> u64 {
-        self.bytes_written
+    pub(crate) fn blob_offset(&self) -> u64 {
+        self.blob_offset
     }
 
     pub(crate) fn header_checksum(&self) -> u32 {
         self.header_checksum
     }
 }
 
+struct RecordWritableDataCreator {
+    record: PartiallySerializedRecord,
+    len: u64,
```

Comment:
I think, it is better to calculate `len` on the fly in the `len()` function, not to store it here. This is because the overhead of copying of this additional 8 bytes field can be larger than just calculating the value on the fly